diff --git a/home/LoBo2_Razno/ESP32/Lua-RTOS-ESP32-ORIG/components/lua_rtos/Lua/modules/net.c b/home/LoBo2_Razno/ESP32/Lua-RTOS-ESP32/components/lua_rtos/Lua/modules/net.c
index 2f1de35..b435480 100644
--- a/home/LoBo2_Razno/ESP32/Lua-RTOS-ESP32-ORIG/components/lua_rtos/Lua/modules/net.c
+++ b/home/LoBo2_Razno/ESP32/Lua-RTOS-ESP32/components/lua_rtos/Lua/modules/net.c
@@ -40,6 +40,7 @@
 #include "net_spi_eth.inc"
 #include "net_service_sntp.inc"
 #include "net_service_http.inc"
+#include "net_service_curl.inc"
 
 #include <stdint.h>
 #include <stdio.h>
@@ -234,6 +235,7 @@ static const LUA_REG_TYPE service_map[] = {
 #if LUA_USE_HTTP
 	{ LSTRKEY( "http" ), LROVAL ( http_map ) },
 #endif
+	{ LSTRKEY( "curl" ), LROVAL ( curl_map ) },
 };
 
 static const LUA_REG_TYPE net_map[] = {
diff --git a/home/LoBo2_Razno/ESP32/Lua-RTOS-ESP32-ORIG/components/lua_rtos/lwip/socket.c b/home/LoBo2_Razno/ESP32/Lua-RTOS-ESP32/components/lua_rtos/lwip/socket.c
index 74b2d7b..d1fedc8 100644
--- a/home/LoBo2_Razno/ESP32/Lua-RTOS-ESP32-ORIG/components/lua_rtos/lwip/socket.c
+++ b/home/LoBo2_Razno/ESP32/Lua-RTOS-ESP32/components/lua_rtos/lwip/socket.c
@@ -51,6 +51,7 @@
 #include <errno.h>
 
 #if USE_NET_VFS
+int _curl_socket = 0;
 #define fd_to_socket(fd) (fd & ((1 << CONFIG_MAX_FD_BITS) - 1))
 #else
 #define fd_to_socket(fd) fd
@@ -243,7 +244,7 @@ int __wrap_lwip_socket(int domain, int type, int protocol) {
 	int s = __real_lwip_socket(domain, type, protocol);
 
 #if USE_NET_VFS
-	if (s != -1) {
+	if ((s != -1) && (!_curl_socket)) {
 		char device[15];
 
 		snprintf(device, sizeof(device), "/dev/socket/%d", s);
